src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py

```diff
@@ -22,8 +22,18 @@
                            'Exception will be thrown if free instances exceed one.')
 register_cli_argument('iot hub create', 'unit', help='Units in your IoT Hub.', type=int)
 
+# Arguments for 'iot hub show-connection-string'
+register_cli_argument('iot hub show-connection-string', 'name', options_list=('--name', '-n'),
+                      help='Target IoT Hub name.')
+register_cli_argument('iot hub show-connection-string', 'policy_name',
+                      help='The access policy you choose to use. Default is iothubowner.')
 
 # Arguments for 'iot device create'
 register_cli_argument('iot device create', 'hub', help='Target IoT Hub name.')
 register_cli_argument('iot device create', 'device_id',
                       options_list=('--device-id', '-d'), help='Device Id.')
+
+# Arguments for 'iot device show-connection-string'
+register_cli_argument('iot device show-connection-string', 'hub', help='Target IoT Hub name.')
+register_cli_argument('iot device show-connection-string', 'device_id',
```

Comment:
You could also add a completer here for hub and possible device_id.  In general we try to provide value completion for all parameters for things that exist. It usually means you'll have a generic alias like so:
```Python
register_cli_argument('iot', 'hub_name', options_list=('--name', '-n'), completer=get_hub_name_list)
register_cli_argument('iot hub create', 'hub_name', completer=None)
```
This way the tab completion is enabled generally (as long as the parameter is named hub_name) but specifically suppressed just for the create command.